File: Java/Language/Runtime/ReflectionTest1/Program1.java
class Program {

	public static void printAsXml(Item info) {
		System.out.println("<Item>");
		System.out.printf("  <name>%s</name>%n", info.name());
		System.out.printf("  <brand>%s</brand>%n", info.brand());
		System.out.println("</Item>");
		System.out.println();
	}

	public static void printAsXml(Customer info) {
		System.out.println("<Customer>");
		System.out.printf("  <id>%s</id>%n", info.id());
		System.out.printf("  <purchase>%s</purchase>%n", info.purchase());
		System.out.printf("  <rating>%s</rating>%n", info.rating());
		System.out.println("</Customer>");
		System.out.println();
	}

	public static void main(String[] args) {
		printAsXml(new Item("cpu", "intel"));
		printAsXml(new Item("ssd", "samsung"));
		printAsXml(new Customer("Alex", 45000, 3));
	}
}

File: dotnet/Language/Runtime/ReflectionTest1/DemoApp/Program1.cs
class Program
{
    private static void PrintAsXml(Item info)
    {
        Console.WriteLine("<Item>");
        Console.WriteLine("  <Name>{0}</Name>", info.Name);
        Console.WriteLine("  <Brand>{0}</Brand>", info.Brand);
        Console.WriteLine("</Item>");
        Console.WriteLine();
    }

    private static void PrintAsXml(Customer info)
    {
        Console.WriteLine("<Customer>");
        Console.WriteLine("  <Id>{0}</Id>", info.Id);
        Console.WriteLine("  <Payment>{0}</Payment>", info.Payment);
        Console.WriteLine("  <Rating>{0}</Rating>", info.Rating);
        Console.WriteLine("</Customer>");
        Console.WriteLine();
    }

    public static void Main()
    {
        PrintAsXml(new Item("Printer", "Canon"));
        PrintAsXml(new Item("WebCam", "Philips"));
        PrintAsXml(new Customer("Jack", 32000, 2));
    }
}